### PR TITLE
Detect late-customization and homescreens support (bug 1219285)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1219285

### CODE REVIEW

File: package/iframe/js/features.js [hunk 1]
```diff
@@ -135,15 +135,72 @@ function generateFeatureProfile(features, navigator) {
     });
 }
 
-function checkForWebExtensions() {
-    return navigator.hasFeature('web-extensions').then(function(value) {
-        return value;
+/*
+ * Utility function that converts an object to an array of objects.
+ * Used because Promise.all() does not accept a litteral object.
```
Comment (chuckharmston) [inline package/iframe/js/features.js]: > literal

File: package/iframe/tests/features.test.js [hunk 2]
```diff
@@ -138,8 +138,38 @@ describe('features', function() {
...
+            });
+        };
+        features.checkForExtraFeatures(navigator).then(function(features) {
+            assert.deepEqual(features, {addonsEnabled: true, homescreensEnabled: true, lateCustomizationEnabled: true});
+            done();
+        });
+    });
 });
 
+
```
Comment (chuckharmston) [inline package/iframe/tests/features.test.js]: :x:

File: package/iframe/js/main.js [hunk 3]
```diff
@@ -65,11 +65,18 @@ function buildQS(profile, hasWebExtensions) {
...
     if (profile) {
+        logger.log('Generated profile: ' + profile);
         qs.push('pro=' + encodeURIComponent(profile));
     }
 
-    if (hasWebExtensions) {
-        qs.push('addonsEnabled=true');
+    if (extraFeatures) {
+        var extraFeaturesKeys = Object.keys(extraFeatures);
+        for (var k = 0; k < extraFeaturesKeys.length; k++) {
```
Comment (ngokevin) [inline package/iframe/js/main.js]: forEach?

File: package/iframe/js/features.js [hunk 4]
```diff
@@ -135,15 +135,72 @@ function generateFeatureProfile(features, navigator) {
...
+    var arr = [];
+
+    // function that returns a function to use as the promise callback.
+    var makeArrayEntry = function(key) {
+        return function(value) {
+            return {name: key, value: value};
+        };
+    };
+
+    for (var key in obj) {
```
Comment (ngokevin) [inline package/iframe/js/features.js]: could use ```Object.keys(obj).map```

File: src/media/js/settings_local_hosted.js [hunk 5]
```diff
@@ -14,6 +14,8 @@ define('settings_local',
         potatolytics_enabled: false,
         tracking_enabled: true,
         addonsEnabled: window.location.search.indexOf('addonsEnabled=true') !== -1,
+        homescreensEnabled: window.location.search.indexOf('homescreensEnabled=true') !== -1,
+        lateCustomizationEnabled:  window.location.search.indexOf('lateCustomizationEnabled=true') !== -1,
```
Comment (ngokevin) [inline src/media/js/settings_local_hosted.js]: found the extra whitespace

**Comment by ngokevin** [comment]
r+wc

**Comment by chuckharmston** [comment]
r+wc

**Pull request merged**